twitterpuller/src/test/java/de/brandwatch/minianalytics/twitterpuller/twitter/TweetFetcherTest.java

```diff
@@ -0,0 +1,52 @@
+package de.brandwatch.minianalytics.twitterpuller.twitter;
+
+
+import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
+import de.brandwatch.minianalytics.twitterpuller.model.Resource;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import static org.junit.Assert.*;
+
+import static org.mockito.Mockito.*;
+
+import org.mockito.junit.MockitoJUnitRunner;
+import twitter4j.Status;
+import twitter4j.User;
+
+import java.util.Date;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TweetFetcherTest {
+
+    @Mock
+    Producer producer;
+
+    @Test
+    public void testIfResourceObjectIsGeneratedCorrectly() {
+
+        Status status = mock(Status.class, RETURNS_DEEP_STUBS);
+
+        Date date = new Date();
+
+        when(status.getText()).thenReturn("Hello World");
+        when(status.getUser().getName()).thenReturn("Max Leopold");
+        when(status.getCreatedAt()).thenReturn(date);
+
+        ArgumentCaptor<Resource> argumentCaptor = ArgumentCaptor.forClass(Resource.class);
+
+        TwitterPullerStatusListener twitterPullerStatusListener = new TwitterPullerStatusListener(producer);
```

Comment:
Instead of this, you should be able to do 

```java
@InjectMocks
private TwitterPullerStatusListener listener;
```

on top, unless you have a good reason to have a separate instance per test.